### ISSUE TITLE
Is there a way to lock NFT for staking?

### ISSUE THREAD

**Issue by akegaviar** [akegaviar/ton-foo#456]
I am looking for NFT staking tutorial but couldn't found.Like other blockchain i want to create a staking platform for my client and they have collection of NFT on TON network.I search everyware in the docs but can't see the tutorial of staking.If anyone know that how to lock NFT in TON network then tell me please.I found that there is multisign wallet and wallet is also contract and NFT also contract, so can we have to create wallet and multisign with user and our key and store NFT for staking.I want that user have NFT after staking but user can not transfer or stake, inshort NFT will be freez for some time.Is that possible in TON blockchain?

## Answer

Hey, in general, there is no TEP (TON Enhancement Proposal) defined for what constitutes "Standard Staking for NFT."

As the contributor who implemented the NFT standard in the Tact language, you can easily set the status on your stake by creating a new contract and designating it as the new owner of the NFT.

Alternatively, you can change the status within the NFT item itself, and then add a `require` statement to limit the `Transfer` method for the NFT item.

In summary, the way to implement this is quite easy and flexible.

I may record a tutorial for this in the near future.

Finally, I'm guessing you tried to complete the Footstep task here ([https://github.com/ton-society/ton-footsteps/issues/295](https://github.com/ton-society/ton-footsteps/issues/295)). I have already pasted a POC (Proof of Concept) code that can run on the testnet. Go check it out! <3

---

[Original](https://answers.ton.org/question/1603114375764250624/is-there-a-way-to-lock-nft-for-staking)
